### PR TITLE
Issue/26506 Provides correct desciption in docstring that get_indexer methods are not yet supported

### LINKED ISSUE #26506: IntervalIndex.get_indexer "method" options are not implemented but documentation implies they are
#### Problem description

`IntervalIndex.get_indexer` returns `NotImplementedError` if `method` is any of the ones shown in the docstring and documentation (except for 'method=None'). I understand that they are not implemented yet, but that is not clear from the docstring.  It presents the 5 available options as if they are implemented. 

Wouldn't it be better to remove `method`, if none of the options are available? Or at least make it more clear in the documentation. 

Documentation: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.IntervalIndex.get_indexer.html

Exception is raised here:
https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/interval.py#L811

### PR DESCRIPTION
**Pull request by timgates42**
- [x] fixes docstring for #26506 until implementation is ready
- [x] 46585 tests passed - none added as documentation change only
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
```
$ git diff upstream/master -u -- "*.py" | flake8 --diff && echo yes
yes
```
- [x] whatsnew entry
Provides correct desciption in docstring that Interval.get_indexer using any method other than default are not yet supported

### CODE REVIEW

**Comment by timgates42** [comment]
I followed the example from get_loc to adjust the documentation, I originally had created a template the each documentation strings used as a basis but was concerned how it would interact with the global documentation dictionary and how it was used. Let me know if you would like any changes.

**Comment by timgates42** [comment]
Oh wow - find the right way to do it in the Contributors guide my apologies for rushing into it without checking the great documentation - I will fix the PR to follow the recomendation in https://pandas-docs.github.io/pandas-docs-travis/development/contributing_docstring.html#sharing-docstrings

**Comment by gfyoung** [comment]
cc @jreback

**Review by WillAyd — CHANGES_REQUESTED** [review]
I think it would be easier to just append a Raises section to the doctoring for IntervalIndex

**Comment by WillAyd** [comment]
cc @jschendel

**Comment by timgates42** [comment]
Thanks - good suggestion will do

**Comment by timgates42** [comment]
Thanks @WillAyd it was much easier - have modified it to append a raises section to the docstring as requested.

**Comment by timgates42** [comment]
Ran into a complication with the plan as the Raises section must be before the Examples in the docstring so hopefully this Substitution technique is the appropriate answer.

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -805,7 +807,15 @@ def get_value(self, series, key):
             loc = self.get_loc(key)
         return series.iloc[loc]
 
-    @Appender(_index_shared_docs['get_indexer'] % _index_doc_kwargs)
+    @Substitution(dict(_index_doc_kwargs,
+                       **{'raises_section': textwrap.dedent("""\
+        Raises
+        ------
+        raises NotImplementedError if any method other than than the default
```
Comment (TomAugspurger) [inline pandas/core/indexes/interval.py]: Typically this is formatted like

```
Raises
------
ExceptionName
    Condition.
```

Comment (timgates42) [reply pandas/core/indexes/interval.py]: Thanks @TomAugspurger - have adjusted accordingly

File: pandas/core/indexes/interval.py [hunk 2]
```diff
@@ -648,6 +648,8 @@ def _check_method(self, method):
             return
 
         if method in ['bfill', 'backfill', 'pad', 'ffill', 'nearest']:
+            # The documentation warns these methods are not implemented so if
```
Comment (jreback) [inline pandas/core/indexes/interval.py]: I don't think this comment actually adds anything here

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -2788,7 +2789,7 @@ def get_loc(self, key, method=None, tolerance=None):
             positions matches the corresponding target values. Missing values
             in the target are marked by -1.
 
-        Examples
+        %(raises_section)sExamples
```
Comment (jreback) [inline pandas/core/indexes/base.py]: should this not be on a separate line? (maybe remove the blank lines at the end of the template)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/interval.py (hunk 2 above)
Comment (timgates42) [reply pandas/core/indexes/interval.py]: Ok I've dropped it.

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (timgates42) [reply pandas/core/indexes/base.py]: Thanks @jreback that does look a lot cleaner - I've adjusted the substitution so its on its own line.

**Comment by WillAyd** [comment]
@timgates42 I think this looks generally good. Can you post a screenshot of the rendered HTML doc?

**Comment by timgates42** [comment]
> @timgates42 I think this looks generally good. Can you post a screenshot of the rendered HTML doc?

Sure @WillAyd here it is

image

**Comment by WillAyd** [comment]
The Raises section looks off with the asterisks - do you know why that is appearing?

**Comment by jreback** [comment]
ahh, just noticed @WillAyd comment, can you update.

**Comment by timgates42** [comment]
Good catch - according to the sphinx documentation this comes out if you are missing a colon (which I am) - I’ll add and regenerate - I actually copied this from another spot in pandas so presumably that needs fixing too - will find it

**Comment by timgates42** [comment]
Oh interesting I found the function I copied rename_categories in pandas/core/arrays/categorical.py and it the issue can be seen in the published doc http://pandas-docs.github.io/pandas-docs-travis/reference/api/pandas.CategoricalIndex.rename_categories.html?highlight=rename_categories#pandas.CategoricalIndex.rename_categories but the fix is not what I expected as pandas uses numpydoc a varient of sphinx doc and it looks like this raises syntax is common across pandas as is the strange asterisks sequence - http://pandas-docs.github.io/pandas-docs-travis/reference/api/pandas.api.types.union_categoricals.html?highlight=union_categoricals#pandas.api.types.union_categoricals

**Comment by timgates42** [comment]
I do get these warnings when producing the documentation which is the next lead to track down...
```
home/tgates/3rdparty/pandas/pandas/core/indexes/interval.py:docstring of pandas.IntervalIndex.get_indexer:54: WARNING: Inline strong start-string without end-string.
WARNING: toctree references unknown document 'reference/api/pandas.RangeIndex.start'
WARNING: toctree references unknown document 'reference/api/pandas.RangeIndex.stop'
WARNING: toctree references unknown document 'reference/api/pandas.RangeIndex.step'
WARNING: toctree references unknown document 'reference/api/pandas.Series.sparse'
```

**Review by WillAyd — APPROVED** [review]
@timgates42 thanks for the research. Given this is a pre-existing issue across the code base I'm OK with this PR as is and would take a follow up PR to fix the asterisks.

Of course up to @jreback as well

**Comment by timgates42** [comment]
Oh just spotted I had doubled up than and it was `any method other than than the default`, I'll fix it.

**Comment by simonjayhawkins** [comment]
> Given this is a pre-existing issue across the code base I'm OK with this PR as is and would take a follow up PR to fix the asterisks.

without checking explicitly, it may be related to #26058, there seem to be a few issues regarding formatting. The dev docs at http://pandas-docs.github.io/pandas-docs-travis are being built on Travis using Sphinx v2.0.1

**Comment by timgates42** [comment]
Oh thanks for the heads up @simonjayhawkins I can confirm I built locally using sphinx v2.0.1 let me check 1.8.5 to see if it looks better.

**Comment by timgates42** [comment]
Rolling back to sphinx 1.8.5 didn't solve the issue, I'll continue investigation. I've setup a separate repo at https://github.com/timgates42/numpydoc-demo to run some testing of numpydoc to see if I can reproduce the issue. If I solve it then I'm happy to send another PR as its unrelated to the problem being addressed here. Let me know if anyone disagrees or wants a change to this PR otherwise assuming all is good please merge it if this is appropriate - Thanks.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @timgates42 

if you can create an issue for the astericks thing (you can just refer back to the comments here), and a PR to fix would be great!

**Pull request merged**